app/src/main/java/com/lagradost/cloudstream3/utils/BiometricAuthenticator.kt

```diff
@@ -0,0 +1,177 @@
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_STRONG
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_WEAK
+import androidx.biometric.BiometricManager.Authenticators.DEVICE_CREDENTIAL
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import com.lagradost.cloudstream3.CommonActivity.showToast
+import com.lagradost.cloudstream3.MainActivity
+import com.lagradost.cloudstream3.R
+
+object BiometricAuthenticator {
+
+    private const val MAX_FAILED_ATTEMPTS = 3
+    private var failedAttempts = 0
+    const val TAG = "cs3Auth"
+
+    private lateinit var biometricManager: BiometricManager
+    lateinit var biometricPrompt: BiometricPrompt
+    lateinit var promptInfo: BiometricPrompt.PromptInfo
+
+    var authCallback: BiometricAuthCallback? = null // listen to authentication success
+
+    private fun initializeBiometrics(activity: Activity) {
+        val executor = ContextCompat.getMainExecutor(activity)
+        biometricManager = BiometricManager.from(activity)
+
+        if (!::promptInfo.isInitialized) {
+            initBiometricPrompt(
+                activity as AppCompatActivity,
+                R.string.biometric_authentication_title,
+                false
+            )
+        }
+
+        biometricPrompt = BiometricPrompt(
+            activity as AppCompatActivity,
+            executor,
+            object : BiometricPrompt.AuthenticationCallback() {
+
+                override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+                    super.onAuthenticationError(errorCode, errString)
+                    showToast("$errString", LENGTH_SHORT)
+                    Log.i(TAG, "$errorCode")
+                    activity.finish()
+                    failedAttempts++
+                    if (failedAttempts >= MAX_FAILED_ATTEMPTS) {
+                        failedAttempts = 0
+                        activity.finish()
+                    } else {
+                        failedAttempts = 0
+                        MainActivity().finish()
+                    }
+                }
+
+                override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                    super.onAuthenticationSucceeded(result)
+                    failedAttempts = 0
+                    authCallback?.onAuthenticationSuccess()
+                }
+
+                override fun onAuthenticationFailed() {
+                    super.onAuthenticationFailed()
+                    failedAttempts++
+                    if (failedAttempts >= MAX_FAILED_ATTEMPTS) {
+                        failedAttempts = 0
+                        activity.finish()
+                    }
+                }
+            })
+    }
+
+    private fun initBiometricPrompt(
+        activity: Activity,
+        title: Int,
+        setDeviceCred: Boolean,
```

Comment:
Please add a comment or something detailing what this is, since it is not obvious based on the name nor usage

Reply:
ok, its the main builder of the prompt